contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -388,10 +422,41 @@ abstract contract PoSValidatorManager is
         });
     }
 
-    /**
-     * @notice See {INativeTokenStakingManager-initializeValidatorRegistration}.
-     * Begins the validator registration process. Locks the provided native asset in the contract as the stake.
-     */
+    function endDelegationCompletedValidator(bytes32 delegationID) external nonReentrant {
+        PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
+
+        Delegator memory delegator = $._delegatorStakes[delegationID];
+        bytes32 validationID = delegator.validationID;
+        Validator memory validator = getValidator(validationID);
+
+        // Ensure the validation is completed
+        require(
+            validator.status == ValidatorStatus.Completed,
+            "PoSValidatorManager: validation not completed"
+        );
+
+        require(
```

Comment:
In the case where the validator is ended after the call to `initializeAddDelegator` but before `completeAddDelegator`, is the expected flow for the delegator user to still call `completeAddDelegator`?

That may not be possible if the validator has ended, since the P-Chain won't be willing to sign the `SubnetValidatorWeightUpdated` messages any longer. We may want to allow the delegator status to be `PendingAdded` here also to support this case.

Reply:
Yeah, they could exit through `completeDelegatorRegistration` without attaching a warp message, but also letting them exit through this function is a little more clear.

Reply:
Ah, @cam-schultz pointed out that in `completeDelegatorRegistration`, the warp message isn't used at all in the case that the validator has already been completed.